druntime/src/core/internal/spinlock.d

```diff
@@ -53,9 +53,9 @@ shared struct SpinLock
         import core.time;
         if (k < pauseThresh)
             return core.atomic.pause();
-        else if (k < 32)
+        else // if (k < 32)
```

Comment:
A sleep fallback for unexpectedly long contentions still seems reasonable.
Maybe simply raise the threshold instead?